aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/UpdateHandler.java

```diff
@@ -79,6 +111,18 @@
             .progress();
     }
 
+    private boolean willAddUpstreams(final ResourceModel desiredModel, final ResourceModel prevModel) {
+        return prevModel.getUpstreams() == null && desiredModel.getUpstreams() != null;
+    }
+
+    private boolean willNotUpdateUpstreams(final ResourceModel desiredModel, final ResourceModel prevModel) {
+        return Objects.equals(prevModel.getUpstreams(), desiredModel.getUpstreams());
```

Comment:
same comment as above, are these lists or sets? if lists, we need to do a better "equals" checker